deploy/jbossas/modeshape-jbossas-subsystem/src/main/resources/schema/modeshape_3_0.xsd

```diff
@@ -574,7 +583,25 @@
             </xs:annotation>
         </xs:attribute>
     </xs:complexType>
-  
+
+    <xs:complexType name="file-persistence">
+        <xs:attribute name="path" type="xs:string" use="optional" default="${jboss.server.data.dir}/modeshape">
+            <xs:annotation>
+                <xs:documentation>
+                    The absolute path on the FS where the repository data should be persisted. If not specified,
+                    defaults to ${jboss.server.data.dir}/modeshape/${repository.name}
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
```

Comment:
the AS config gets translated by our AS services to JSON; this in turn does have support for env variables.

Considering that the [full list](https://docs.jboss.org/author/display/WFLY8/Command+line+parameters) of variables covers pretty much all of the "interesting" server dirs, it should be enough for our needs.